src/app/component/PushNotification/PushNotification.component.js

```diff
@@ -0,0 +1,159 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import { GRANTED, DENIED, DEFAULT } from '../../store/Notification/Notification.reducer';
+
+class PushNotification extends Component {
+    constructor(props) {
+        super(props);
+
+        this.granted = props.grantType === GRANTED;
+        this.windowFocus = document.visibilityState === 'visible';
+    }
+
+    componentDidMount() {
+        window.addEventListener('blur', this.onWindowBlur);
+        window.addEventListener('focus', this.onWindowFocus);
+
+        const {
+            forceAsk,
+            supported,
+            grantType
+        } = this.props;
+
+        if (supported === false) {
```

Comment:
I prefer inline if if there is only one line inside. But it depends on complexity. Here, I would inline.